java/fury-core/src/main/java/org/apache/fury/reflect/TypeToken.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.fury.reflect;
```

Comment:
does this class borrow any code from https://github.com/google/guava/blob/master/guava/src/com/google/common/reflect/TypeToken.java ?

If so, we need to keep the license header from the guava class file and change the Fury license to mention that we have guava code included in fury.

If this is a pure clean room implementation, great. But if you there is any doubt, then we should include the guava license details.

Reply:
Yes, most of the code here is borrowed from guava's `TypeToken`. Let me update the license header and the Fury license.

Thanks for pointing this out.

Reply:
take the apache header off this file then too - if most of the code is actually borrowed from gson

Reply:
Updated the file header and Fury license.